src/components/DataDisplay/Chip/Chip.tsx

```diff
@@ -1,24 +1,221 @@
-import { forwardRef } from 'react';
+import {
+  Children as ReactChildren,
+  forwardRef,
+  Fragment,
+  HTMLAttributes,
+  isValidElement,
+  ReactElement,
+  ReactNode,
+} from 'react';
 
-import { ChipProps as EdsChipProps } from '@equinor/eds-core-react';
+import { Icon } from '@equinor/eds-core-react';
+import { IconData } from '@equinor/eds-icons';
 
-import { StyledChip } from './Chip.styles';
+import { InteractiveChipStyle, ReadOnlyChipStyle } from './Chip.styles';
 
-type ChipProps = EdsChipProps & {
-  readonly?: boolean;
-};
+// Base type for common properties
+export interface BaseChipProps {
+  children: ReactNode;
+  disabled?: boolean;
+  onDelete?: (event: unknown) => void;
+  variant?:
+    | 'default'
+    | 'active'
+    | 'warning'
+    | 'warning-active'
+    | 'error'
+    | 'error-active';
+  onClick?: (event: unknown) => void;
+  leadingIconData?: IconData;
+}
+
+// Type for ReadOnlyChipStyle (div)
+export type ReadOnlyChipProps = BaseChipProps & HTMLAttributes<HTMLDivElement>;
+
+// Type for InteractiveChipStyle (button)
+export type InteractiveChipProps = BaseChipProps &
+  HTMLAttributes<HTMLButtonElement>;
+
+type ReactElementWithChildren = ReactElement<{ children?: ReactNode }>;
+
+// Define the ReadOnlyChip component
+const ReadOnlyChip = forwardRef<HTMLDivElement, ReadOnlyChipProps>(
+  (props, ref) => {
+    const { children, disabled, leadingIconData, ...otherReadOnlyProps } =
+      props;
+
+    // ReadOnlyChip only needs to handle the div related props and children,
+    return (
+      <ReadOnlyChipStyle
+        {...otherReadOnlyProps}
+        className={disabled ? 'disabled' : ''}
+        ref={ref}
+      >
+        <div className="content">
+          {leadingIconData && (
+            <div className="leading">
+              <Icon data={leadingIconData} size={16} />
+            </div>
+          )}
+          {children}
+        </div>
+      </ReadOnlyChipStyle>
+    );
+  }
+);
+ReadOnlyChip.displayName = 'ReadOnlyChip';
+
+// Define the InteractiveChip component
+const InteractiveChip = forwardRef<HTMLButtonElement, InteractiveChipProps>(
```

Comment:
ReadOnly and InteractiveChip should be placed in separate files (`ReadOnlyChip.tsx` and `InteractiveChip.tsx`)